File: src/atr_pkg/scripts/ma_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry

def chao_callback(msg):
    rospy.loginfo(msg.grade)
    rospy.loginfo(str(msg.star)+"星")
    rospy.loginfo(msg.data)

def yao_callback(msg):
    rospy.logwarn(msg.grade)
    rospy.logwarn(str(msg.star)+"星")
    rospy.logwarn(msg.data)


if __name__ == "__main__":
    rospy.init_node("ma_node")
    rospy.logwarn("this is py_ma_node~")

    sub = rospy.Subscriber("chao_topic", Carry, queue_size=10, callback=chao_callback)
    sub2 = rospy.Subscriber("yao_topic", Carry, queue_size=10, callback=yao_callback)

    while not rospy.is_shutdown():
        rospy.spin()

# def chao_callback(msg):
#     rospy.loginfo(msg.data)

# def yao_callback(msg):
#     rospy.logwarn(msg.data)


# if __name__ == "__main__":
#     rospy.init_node("ma_node")
#     rospy.logwarn("this is py_ma_node~")

#     sub = rospy.Subscriber("chao_topic", String, queue_size=10, callback=chao_callback)
#     sub2 = rospy.Subscriber("yao_topic", String, queue_size=10, callback=yao_callback)

#     while not rospy.is_shutdown():
#         rospy.spin()

File: src/atr_pkg/src/ma_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <qq_msgs/Carry.h>

void chao_callback(qq_msgs::Carry msg)
{
    ROS_INFO(msg.grade.c_str());
    ROS_INFO("%d 星", msg.star);
    ROS_INFO(msg.data.c_str());
}

void yao_callback(qq_msgs::Carry msg)
{
    ROS_WARN(msg.grade.c_str());
    ROS_WARN("%d 星", msg.star);
    ROS_WARN(msg.data.c_str());
}

int main(int argc, char *argv[])
{
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "ma_node");
    printf("this is ma_node~\n");

    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("chao_topic", 10, chao_callback);
    ros::Subscriber sub2 = nh.subscribe("yao_topic", 10, yao_callback);

    while (ros::ok())
    {
        ros::spinOnce();
    }
    
    return 0;
}


// void chao_callback(std_msgs::String msg)
// {
//     ROS_INFO(msg.data.c_str());
// }

// void yao_callback(std_msgs::String msg)
// {
//     ROS_WARN(msg.data.c_str());
// }

// int main(int argc, char *argv[])
// {
//     ros::init(argc, argv, "ma_node");

//     ros::NodeHandle nh;
//     ros::Subscriber sub = nh.subscribe("chao_topic", 10, chao_callback);
//     ros::Subscriber sub2 = nh.subscribe("yao_topic", 10, yao_callback);

//     while (ros::ok())
//     {
//         ros::spinOnce();
//     }
    
//     return 0;
// }
